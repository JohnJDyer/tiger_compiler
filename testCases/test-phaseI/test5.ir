    assign, 0b, 0,
    assign, 0a, 0,
main:
    assign, $t0, 0,
    brneq, 0a, 0b, if_label0
    assign, $t0, 1,
if_label0:
    breq, $t0, 0, if_label1
    add, 0b, 2, $t1
    assign, 0a, $t1,
    goto, if_label2, ,
if_label1:
    assign, 0a, 2,
if_label2:
    call, printi, 0a
    return, , ,
