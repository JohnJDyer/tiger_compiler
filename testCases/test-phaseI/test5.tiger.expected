	assign, a, 0, 
	assign, b, 0, 
main:
	assign, $t0, 1, 
	breq, a, b, compare_label0
	assign, $t0, 0, 
compare_label0:
	breq, $t0, 0, if_label1
	add, b, 2, $t1
	assign, a, $t1, 
	goto, if_label2, , 
if_label1:
	assign, a, 2, 
if_label2:
	call, printi, a
	return, , , 
